module_utils/network_lsr/argument_validator.py

```diff
@@ -1553,6 +1553,8 @@ def get_default_bond(self):
 
 
 class ArgValidator_DictInfiniband(ArgValidatorDict):
+    DEFAULT_PKEY = -1
```

Comment:
as said before, when talking about the playbook/YAML syntax, the physical interface should primarily be represented by not specifying the `p_key` property (only accept `-1` as alternative).

ArgValidator takes the playbook, and normalizes it to something that is most useful for internal use. It does not necessarily need to use the same representation as in yaml. Rather, what is useful. As such, if the value is missing in YAML, ArgValidator might normalize that to `-1`. Or it might normalize it to None. I don't see any advantage of using -1 here. In particular, if you now feel the urge to add a DEFAULT_PKEY alias only to represent an unset value. We already have a well-known, commonly used value to represent missing values: `None`.

One argument could be, tha if you use -1, then you could always call
```
                 s_infiniband.set_property(
                     NM.SETTING_INFINIBAND_P_KEY, connection["infiniband"]["p_key"]
                 )
```
(which is true, as far as libnm is concerned). But the code doesn't even make use of that. It explicitly checks:
```
            if (
                connection["infiniband"]["p_key"]
                != ArgValidator_DictInfiniband.DEFAULT_PKEY
            ):
                s_infiniband.set_property(
                    NM.SETTING_INFINIBAND_P_KEY, connection["infiniband"]["p_key"]
                )
```


Only because libnm does something, it does not mean that this is also the best representation of the setting -- neither in playbook/YAML nor for ArgValidator.   Usually yes (because what libnm does often makes sense already, and because we might be familiar with that), but not always.

Reply:
if you were to leave it at `-1`, I wouldn't say anything. After all, I added that (without thinking properly).
But now building on top of that wrong choice, by adding `ArgValidator_DictInfiniband.DEFAULT_PKEY`, seems not good.

Reply:
also, this commit fixes one actual bug (the `!=` vs. `==`). It's a one-line fix (even a one-character fix).

Instead, this commit is +25/-5 lines of unrelated change. The bugfix is important and it's almost impossible to see, please create separate commits.

Reply:
> as said before, when talking about the playbook/YAML syntax, the physical interface should primarily be represented by not specifying the `p_key` property (only accept `-1` as alternative).
> 
> ArgValidator takes the playbook, and normalizes it to something that is most useful for internal use. It does not necessarily need to use the same representation as in yaml. Rather, what is useful. As such, if the value is missing in YAML, ArgValidator might normalize that to `-1`. Or it might normalize it to None. I don't see any advantage of using -1 here. In particular, if you now feel the urge to add a DEFAULT_PKEY alias only to represent an unset value. We already have a well-known, commonly used value to represent missing values: `None`.
> 
> One argument could be, tha if you use -1, then you could always call
> 
> ```
>                  s_infiniband.set_property(
>                      NM.SETTING_INFINIBAND_P_KEY, connection["infiniband"]["p_key"]
>                  )
> ```
> 
> (which is true, as far as libnm is concerned). But the code doesn't even make use of that. It explicitly checks:
> 
> ```
>             if (
>                 connection["infiniband"]["p_key"]
>                 != ArgValidator_DictInfiniband.DEFAULT_PKEY
>             ):
>                 s_infiniband.set_property(
>                     NM.SETTING_INFINIBAND_P_KEY, connection["infiniband"]["p_key"]
>                 )
> ```
> 
> Only because libnm does something, it does not mean that this is also the best representation of the setting -- neither in playbook/YAML nor for ArgValidator. Usually yes (because what libnm does often makes sense already, and because we might be familiar with that), but not always.

Thanks for the explanation, I will change the default value of `p_key` into  `None`.

Reply:
> also, this commit fixes one actual bug (the != vs. ==). It's a one-line fix (even a one-character fix).

Is this fix done? Please make it a separate commit.

Reply:
I changed the order of the commits.